### PR TITLE
Add custom constructor section for oggbundle.

### PR DESCRIPTION
**Pull request by deiferni**
See https://github.com/4teamwork/opengever.core/issues/2308.

This PR adds a transmogrifier section that constructs instances.

*CL will follow in a later PR*

### CODE REVIEW

File: opengever/setup/sections/constructor.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            if fti is None:
+                raise InvalidType(portal_type)
+
+            try:
+                # we need the title sometimes to auto-generate ids
+                # XXX maybe be a bit more intelligent here and set all required
+                # fields while constructing?
+                obj = createContentInContainer(
+                    context, portal_type,
+                    title_de=item.get(u'title_de'),
```
Comment (phgross) [inline opengever/setup/sections/constructor.py]: Does this attribute has impact for object without the TranslatedTitle behavior? Maybe extending the tests would be a good thing ...

Comment (deiferni) [inline opengever/setup/sections/constructor.py]: hmm right, should be extra careful here, update implementation to check for `ITranslatedTitle` behaviour and only then user the language dependent fields.

**Comment by deiferni** [comment]
@phgross updated

File: opengever/setup/sections/constructor.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            fti = self.ttool.getTypeInfo(portal_type)
+            if fti is None:
+                raise InvalidType(portal_type)
+
+            try:
+                # we need the title sometimes to auto-generate ids
+                # XXX maybe be a bit more intelligent here and set all required
+                # fields while constructing?
+                if self.has_translated_title(fti):
+                    title_keys = (u'title_de', u'title_fr')
```
Comment (phgross) [inline opengever/setup/sections/constructor.py]: Why not using the `TRANSLATED_TITLE_NAMES`?

Comment (deiferni) [inline opengever/setup/sections/constructor.py]: 🙈  that's why i've refactored it in the first place, thanks ...

**Comment by deiferni** [comment]
@phgross let's try again 😉  - updated

**Pull request merged**